Sources/NIO/EventLoop.swift

```diff
@@ -823,6 +823,18 @@ extension EventLoopGroup {
 typealias ThreadInitializer = (Thread) -> Void
 
 /// An `EventLoopGroup` which will create multiple `EventLoop`s, each tied to its own `Thread`.
+///
+/// The effect of initializing a `MultiThreadedEventLoopGroup` is to spawn `numberOfThreads` fresh threads which will
+/// all run their own `EventLoop`. Those threads will not be shut down until `shutdownGracefully` or
+/// `syncShutdownGracefully` is called.
+///
+/// - note: It's good style to call `MultiThreadedEventLoopGroup.shutdownGracefully` or
+///         `MultiThreadedEventLoopGroup.syncShutdownGracefully` when you no longer need this `EventLoopGroup`. In
+///         many cases that is just before your program exits.
+/// - warning: Unit tests often spawn one `MultithreadedEventLoopGroup` per unit test to force isolation between the
+///            tests. In those cases it's important to shut the `MultithreadedEventLoopGroup` down at the end of the
+///            test. A good place to start a `MultithreadedEventLoopGroup` is the `setUp` method of your `XCTestCase`
+///            subclass, a good place to shut it down is the `tearDown` method.
```

Comment:
Nit: a number of the references to the name of this class in this block have incorrect casing. The correct casing is `MultiThreadedEventLoopGroup`. Mind fixing those up?